plugins/by-name/alpha/default.nix

```
@@ -1,177 +1,113 @@
+        type = "padding";
+        val = 2;
+      }
+      {
+        type = "group";
+        val = [
           {
-            type = "group";
-            val = [
-              {
-                type = "button";
-                val = "  New file";
-                on_press.__raw = "function() vim.cmd[[ene]] end";
-                opts.shortcut = "n";
-              }
-              {
-                type = "button";
-                val = " Quit Neovim";
-                on_press.__raw = "function() vim.cmd[[qa]] end";
-                opts.shortcut = "q";
-              }
-            ];
+            type = "button";
+            val = "  New file";
+            on_press = lib.nixvim.nestedLiteralLua "function() vim.cmd[[ene]] end";
+            opts.shortcut = "n";
           }
           {
-            type = "padding";
-            val = 2;
-          }
-          {
-            type = "text";
-            val = "Inspiring quote here.";
-            opts = {
-              position = "center";
-              hl = "Keyword";
-            };
+            type = "button";
+            val = " Quit Neovim";
+            on_press = lib.nixvim.nestedLiteralLua "function() vim.cmd[[qa]] end";
+            opts.shortcut = "q";
           }
         ];
-      };
+      }
+      {
+        type = "padding";
+        val = 2;
+      }
+      {
+        type = "text";
+        val = "Inspiring quote here.";
+        opts = {
+          position = "center";
+          hl = "Keyword";
+        };
+      }
+    ];
+  };
 
-      opts = lib.nixvim.mkNullOrOption (with types; attrsOf anything) ''
-        Optional global options.
-      '';
+  extraOptions = {
+    theme = lib.mkOption {
+      type =
+        with lib.types;
+        let
+          # TODO: deprecated 2025-10-30, remove after 26.05
+          old = nullOr (maybeRaw str);
+          new = nullOr str;
+        in
+        old // { inherit (new) description; };
+      default = null;
+      example = "dashboard";
+      description = "You can directly use a pre-defined theme.";
     };
   };
 
-  config =
-    let
-      layoutDefined = cfg.layout != [ ];
-      themeDefined = cfg.theme != null;
-    in
-    lib.mkIf cfg.enable {
-      extraPlugins = [ cfg.package ];
-
-      assertions = lib.nixvim.mkAssertions "plugins.alpha" [
-        {
-          assertion = themeDefined || layoutDefined;
-          message = ''
-            You have to either set a `theme` or define some sections in `layout`.
-          '';
-        }
-        {
-          assertion = !(themeDefined && layoutDefined);
-          message = ''
-            You can't define both a `theme` and custom options.
-            Set `plugins.alpha.theme = null` if you want to configure alpha manually using the `layout` option.
-          '';
-        }
-      ];
+  callSetup = false;
+  extraConfig = cfg: opts: {
+    assertions = lib.nixvim.mkAssertions "plugins.alpha" {
+      assertion = !(cfg.theme ? __raw);
```

Probably better to do:


```suggestion
      assertion = cfg.theme != null -> builtins.isString cfg.theme);
```

Because maybeRaw also allows nixvim's lua-inline type.